src/HIE/Bios/Config.hs

```diff
@@ -16,33 +17,25 @@ data CradleConfig = Cabal { component :: Maybe String }
                   | Obelisk
                   | Bios { prog :: FilePath }
                   | Default
-                  deriving (Generic, Show)
-
-instance Interpret CradleConfig
+                  deriving (Show)
+
+instance FromJSON CradleConfig where
+    parseJSON (Object (Map.toList -> [(key, val)]))
+        | key == "cabal" = case val of
+            Object x | Just (String v) <- Map.lookup "component" x -> return $ Cabal $ Just $ T.unpack v
+            _ -> return $ Cabal Nothing
```

Comment:
This is going to be confusing because it will accept `cabal: garbage`. Something to change in future.